lineapy/graph_reader/artifact_collection.py

```diff
@@ -49,7 +49,9 @@ def __init__(
             List[str], List[Tuple[str, int]], List[Union[str, Tuple[str, int]]]
         ],
         input_parameters: List[str] = [],
-        reuse_pre_computed_artifacts: List[Union[str, Tuple[str, int]]] = [],
+        reuse_pre_computed_artifacts: Union[
+            List[str], List[Tuple[str, int]], List[Union[str, Tuple[str, int]]]
```

Comment:
this seems unnecessary. what was there before should've accounted for this. do you have a sample that didnt work?

Reply:
What was before does not support the following signature in `to_pipeline()`: 

```python
def to_pipeline(
    ...
    reuse_pre_computed_artifacts: List[str] = [],
    ...
) -> Path:
```

Reply:
I think we need new type like `ArtifactDefinition` to clean all these typing. I personally want to do this when we add more attributes(like namespace/project/user ...) to the Artifact.